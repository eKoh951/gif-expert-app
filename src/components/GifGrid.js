import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

	const [images, setImages] = useState([]);

	// The useEffect hook helps to execute code only in certain changes of the component
	useEffect( () => {
		getGifs( category )
			// .then( imgs => setImages( imgs ));
			.then( setImages );
		// If we don't provide any dependency (inside the array), the code is executed once
	}, [ category ]);

	// getGifs();

	return (
		<div className="card-grid">
			<h3>{ category }</h3>
			{
				images.map( img => (
					<GifGridItem 
						key={ img.id }
						{ ...img }
					/>
				))
			}
		</div>
	)
}
