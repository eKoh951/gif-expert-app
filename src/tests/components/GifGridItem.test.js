import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';

describe('Prueba de componente GifGridItem', () => {
	let wrapper;

	beforeAll(() => {
		const img = {
			id: 'mAJxVnLmZcZMzhyPvO',
			title: 'One Punch Man GIF by memecandy',
			url: 'https://media1.giphy.com/media/mAJxVnLmZcZMzhyPvO/giphy.gif?cid=91b2ca0cphho7fz2liinkjwa9hugoqx2l3ka56bmxa42q50v&rid=giphy.gif'
		}
		
		wrapper = shallow(
		<GifGridItem
			key={ img.id }
			{ ...img }
		/>);
	})

	test('debe de mostrar el componente correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	})
})