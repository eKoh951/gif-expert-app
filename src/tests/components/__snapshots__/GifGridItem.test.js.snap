// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de componente GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="One Punch Man GIF by memecandy"
    src="https://media1.giphy.com/media/mAJxVnLmZcZMzhyPvO/giphy.gif?cid=91b2ca0cphho7fz2liinkjwa9hugoqx2l3ka56bmxa42q50v&rid=giphy.gif"
  />
  <p>
    One Punch Man GIF by memecandy
  </p>
</div>
`;
